core/pom.xml

```diff
@@ -35,7 +35,7 @@
         <nodejs.version>v16.13.1</nodejs.version>
         <runtime.assembly>none</runtime.assembly>
         <javac.target>11</javac.target>
-        <argLine>-Dfile.encoding=UTF-8</argLine>
+        <argLine>-ea -Dfile.encoding=UTF-8</argLine>
```

Comment:
I have spotted something while testing, will show you.

Reply:
Note: this test was shown to be flaky RetryIODispatcherTest.testImportRerunsExceedsRerunProcessingQueueSizeLoop in linux-jdk8

Reply:
Regadring "-ea" flag - yes, I'd like to turn on assertions during test execution in CI .

Reply:
The issue also affects master so it's a pgwire bug we weren't aware of .

Reply:
Converted to draft to fix this  issue found by Miguel.
Turns out qdb returns query data using wrong types in "Extended query" mode when it tries to reuse cached query but fails because metadata is not up to data and query needs re-compiling . 
In this case it returns data as chars in 'Data row' message even though it declared column as int in 'Row description' message.